lib/app/register/register_presenter.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:hnh/domain/repositories/authentication_repository.dart';
+import 'package:hnh/domain/usecases/auth/register_usecase.dart';
+import 'package:hnh/domain/usecases/observer.dart';
+import 'package:flutter/foundation.dart';
+
+class RegisterPresenter {
+
+  AuthenticationRepository _authenticationRepository;
+  RegisterUserCase _registerUserCase;
+
+  Function registerOnComplete;
+  Function registerOnError;
+
+  RegisterPresenter(this._authenticationRepository) {
+    _registerUserCase = RegisterUserCase(_authenticationRepository);
+  }
+
+  void _dispose() {
+    _registerUserCase.dispose();
+  }
+
+  void register({@required String firstName, String lastName, String email, String password}) {
```

Comment:
Add `@required` to the rest of the parameters

Reply:
Changes made